Python/tigre/algorithms/pocs_algorithms.py

```diff
@@ -162,8 +162,7 @@ def run_main_iter(self):
                 dtvg = dtvg * self.alpha_red
 
             self.beta *= self.beta_red
-            c = np.dot(dg_vec.reshape(-1,), dp_vec.reshape(-1,)) / \
-                max(dg * dp, 1e-6)
+            c = np.dot(dg_vec.ravel(), dp_vec.ravel()) / (dg * dp)

```

Comment:
I need to think about this. They do the same thing in practice, but if I dont understand wrong, `ravel` _may_ return a copy, while `reshape(-1,0)` is ensured not to return a copy.  

As usual, if this is the case, we come to the speed vs memory efficiency situation. `ravel` will return a copy if the array is not contiguous in memory, meaning the `dot` would be slower. A copy will make `dot` faster. But then we need a couple of copies of the image, and this can be bad at large scale....

Let me think about this.